src/anndata/_core/merge.py

```diff
@@ -143,11 +143,12 @@ def equal_dask_array(a, b) -> bool:
         return False
     if isinstance(b, DaskArray) and tokenize(a) == tokenize(b):
         return True
-    if isinstance(a._meta, CSMatrix | CSArray):
-        # TODO: Maybe also do this in the other case?
-        return da.map_blocks(equal, a, b, drop_axis=(0, 1)).all()
-    else:
-        return da.equal(a, b, where=~(da.isnan(a) == da.isnan(b))).all()
+    if a.chunksize == b.chunksize:
+        if isinstance(a._meta, CupySparseMatrix | CSMatrix | CSArray):
+            # TODO: Maybe also do this in the other case?
+            return da.map_blocks(equal, a, b, drop_axis=(0, 1)).all()
+        return da.equal(a, b, where=~(da.isnan(a) + da.isnan(b))).all().compute()
```

Comment:
Genuinely don't know how this was ever working with `da.isnan(a) == da.isnan(b)` would basically always return all True, so the `where=~()` would always be False, which meant nothing was checked for equality.  Or am I missing something?

So I changed to `da.isnan(a) + ds.isnan(b)` so that we check where either or b are not nan.  But maybe this check is wrong?